tbx/static_src/javascript/components/youtube-embed.js

```diff
@@ -0,0 +1,79 @@
+import Cookies from 'js-cookie';
+
+/**
+ * YouTubeConsentManager handles the consent for loading YouTube videos on a webpage.
+ */
+class YouTubeConsentManager {
+    static selector() {
+        return '.grid__embed.streamfield__embed.youtube-video-container';
```

Comment:
We never use classes to target JS as this can cause issues if the styling for the page is updated. Instead we use data attrbutes.